scripts/Muon/GUI/Common/plot_widget/plotting_canvas/plotting_canvas_view.py

```diff
@@ -207,24 +207,32 @@ def set_axis_ylimits(self, axis_number, ylims):
     def set_axes_limits(self, xlim, ylim):
         plt.setp(self.fig.axes, xlim=xlim, ylim=ylim)
 
-    def autoscale_y_axes(self):
+    def autoscale_y_axes(self,yerr=0):
         ymin = 1e9
         ymax = -1e9
+
         for axis in self.fig.axes:
             ymin_i, ymax_i = self._get_y_axis_autoscale_limts(axis)
             if ymin_i < ymin:
                 ymin = ymin_i
             if ymax_i > ymax:
                 ymax = ymax_i
 
+        ymax,ymin = ymax + yerr , ymin - yerr
+
         plt.setp(self.fig.axes, ylim=[ymin, ymax])
 
-    def autoscale_selected_y_axis(self, axis_number):
+    def autoscale_selected_y_axis(self, axis_number,yerr = 0):
```

Comment:
this is for a single axis right? So for multiple axis should we not just loop over this function and pass the correct axis?